File: crates/c-api/tests/engine.cc
#include <wasmtime/engine.hh>

#include <gtest/gtest.h>

using namespace wasmtime;

TEST(Engine, Simple) {
  Engine engine;
  engine = Engine(Config());
  engine.increment_epoch();

  Engine engine2 = engine;
  engine2 = Engine();
  engine.is_pulley();
}

File: crates/fuzzing/tests/oom/engine.rs
#![cfg(arc_try_new)]

use wasmtime::{Config, Engine, Result};
use wasmtime_fuzzing::oom::OomTest;

#[test]
fn engine_new() -> Result<()> {
    OomTest::new().test(|| {
        let mut config = Config::new();
        config.enable_compiler(false);
        let _ = Engine::new(&config)?;
        Ok(())
    })
}
